app/web/src/components/IdeViewer/IdeViewer.js

```diff
@@ -14,19 +14,27 @@ import { requestRender } from 'src/helpers/hooks/useIdeState'
 
 extend({ OrbitControls })
 
-function Asset({ stlData }) {
+function Asset({ url, resetLoading, setLoading }) {
   const [loadedGeometry, setLoadedGeometry] = useState()
   const mesh = useRef()
   const ref = useUpdate((geometry) => {
     geometry.attributes = loadedGeometry.attributes
   })
   useEffect(() => {
-    if (stlData) {
-      const decoded = atob(stlData)
+    if (url) {
       const loader = new STLLoader()
-      setLoadedGeometry(loader.parse(decoded))
+      setLoading()
+      loader.load(
+        url,
+        (geometry) => {
+          setLoadedGeometry(geometry)
+          resetLoading()
+        },
+        null,
+        resetLoading
+      )
```

Comment:
Because the asset is not being returned directly, but instead a URL to it, I now need to make sure the loading animation stays on while the asset loads, especially for the stls as they can be large.